teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -84,97 +84,399 @@ that can be shared between them, and running the second Task will be delayed by
   can be run on the same pod as another Task that fails
   - This is to support use cases such as uploading test results, even if the test
     Task failed
     - This requirement is being included because we could choose a solution that doesn't
       address the above use case; for example in PipelineResources, you can have a
       storage "output" but if the steps fail, the "output" pipelineresource will not run
+1. It must be possible for Tasks within a Pipeline that is run in this manner to be able to specify different levels
+  of [hermeticity](https://github.com/tektoncd/community/blob/main/teps/0025-hermekton.md)
+  - For example, a Task that fetching dependencies may need to run in non-hermetic mode, but the Task that actually
+    builds an artifact using these dependencies may need to be hermetic (e.g. 
+1. Information about the status of individually executing Tasks must still be surfaced in a way such that
+   in the `status` of the PipelineRun, the execution of each individual Task (including its results) can be viewed
+   separately
+    - i.e. even if multiple Tasks execute as one pod, the results of each execution should be displayed to the user
+      separately in the `status`
+    - [PipelineRuns currently specify this information in a `taskruns` section ](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#monitoring-execution-status)
+      so if the execution method ends up not being individual `TaskRuns` we may need to either create "fake" taskruns
+      to populate, or create a new section with different status information
 
 ## Design details
 
 TBD - currently focusing on enumerating and examining alternatives before selecting one or more ways forward.
 
 ## Alternatives
 
-Most of these options are not mutually exclusive:
-
-* [Task composition in Pipeline Tasks](#task-composition-in-pipeline-tasks)
-* [Update PipelineResources to use Tasks](#update-pipelineresources-to-use-tasks)
-* [Automatically combine Tasks based on workspace use](#automagically-combine-tasks-based-on-workspace-use)
-* [Introduce scheduling rules to Pipeline](#introduce-scheduling-rules-to-pipeline)
-* [PipelineRun: emptyDir](#pipelinerun-emptydir)
-* [Controller configuration](#controller-level)
-* [Within the Task](#within-the-task)
-* [Remove distinction between Tasks and Pipelines](#remove-distinction-between-tasks-and-pipelines)
-* [Custom Pipeline](#custom-pipeline)
-* [Create a new Grouping CRD](#create-a-new-grouping-crd)
-* [Rely on the Affinity Assistant](#rely-on-the-affinity-assistant)
-* [Custom scheduler](#custom-scheduler)
-* [Support other ways to share data (e.g. buckets)](#support-other-ways-to-share-data-eg-buckets)
-* [Focus on workspaces](#focus-on-workspaces)
-
-Most of the solutions above involve allowing more than one Task to be run in the same pod, and those proposals all share
-the following pros & cons.
+[Implementation options](#implementation-options):
```

Comment:
I really like the strategy of separating implementation options from syntax options here!